app/models/donation.rb

```diff
@@ -5,4 +5,6 @@ class Donation < ActiveRecord::Base
 
   validates :donor, presence: true
   validates :amount, presence: true, numericality: { greater_than: 0 }
+
+  scope :search, -> (keyword) { Donation.joins(:donor).where("name ILIKE ?", "%#{keyword}%") }
```

Comment:
We're already in the `Donation` model, so specifying `Donation` in the scope isn't necessary. 😀 

```
scope :search, -> (keyword) { joins(:donor).where("name ILIKE ?", "%#{keyword}%") }
```